graph-manager-docs/source/federation.mdx

```diff
@@ -127,9 +127,113 @@ server.listen().then(({ url }) => {
 
 Managed configuration is the concept of allowing the gateway to update dynamically in response to service changes. In principle, it's important to separate the reliability of your graph from the reliability of your services. For that reason, we recommend running federation in production using a managed configuration, where the gateway picks up config changes not from introspection, but from a set of files owned by your graph describing its current state.
 
-On `apollo service:push`, the service registry writes configuration files to a cloud file system, stored securely and accessible by your API key. These configuration files detail which federated services are part of the graph, metadata about the federated services, including the `serviceURL` specified in the `push` command, and the partial schema of that federated service. By default, when the Apollo gateway is instantiated with an API key rather than a `serviceList`, it will poll for that managed config to pick up changes and smoothly roll over to the new service configuration, draining in-flight requests while beginning to generate query plans for incoming requests against the new config.
+On `apollo service:push`, the service registry writes configuration files to a cloud file system, stored securely and accessible by your API key.
+These configuration files detail which federated services are part of the graph, metadata about the federated services, including the `serviceURL`
+ specified in the `push` command, and the partial schema of that federated service. By default, when the Apollo gateway is instantiated with an
+  API key rather than a `serviceList`, it will poll for that managed config to pick up changes and smoothly roll over to the new service configuration,
+  draining in-flight requests while beginning to generate query plans for incoming requests against the new config.
+
+#### When to call `service:push`
+
+In general, when calling `service:push`, you should always roll out changes in a way that is backward compatible with existing clients to avoid downtime. Because configuration changes can affect the query planner, it's highly recommended to **call `apollo service:push` after
+all replicas of a federated service have deployed**. This way, you can be certain that only valid operations will be executable against the graph, and
+that operations will not be able to access fields or types that do not yet exist.
+
+In the rare case where a schema change is _not_ backward compatible with your gateway's query planner, you should update your service registry _before_ you deploy your federated services. In particular, schema changes that affect query planning should be done as separate changes to the registry, prior
```

Comment:
I think "not backward compatible with your gateway's query planner" doesn't mean the same thing as "not backward compatible (just, like, as a schema itself)" but this is subtle and should be called out.

Like, it sounds like there are actually two different cases: "a backwards incompatible change that doesn't mess up QP" and "a backwards incompatible change that does mess up QP"? But we only really talk about the latter?

Reply:
also it looks like this file is inconsistent about "backwards" vs "backward" (both are apparently ok, "backward" is more common in the US?)

Reply:
I don't talk about backwards incompatible schema changes because it's pointed out that you should never make those.

Reply:
Uh but we do them all the time.

And if we want our product's opinion to be "never make backward compatible schema changes" then I guess we can literally remove the entire giant schema validation feature where we validate the new schema against historical queries, and instead just reject backwards incompatible changes? Seems like we sure have a ton of infrastructure built around the the concept of backwards-incompatible changes being an OK thing to do if you are convinced there aren't users that will be broken.

Reply:
the point is that those *are* backwards compatible! When I say backwards compatible here, I mean the change itself can be backwards compatible or it can be so by the fact no one is using it. Our docs use the phrase repeatedly to mean that you're validating that your changes _are_ backwards compatible by using validation, but I'm happy to try to be more descriptive here if you think the language can be improved.

Reply:
I haven't had time to fully re review this PR and probably won't. But I just want to register that this seems really weird to me.  To me backwards incompatibility is a property of what the change is, not of whether or not people happen to be still making use of the old behavior.  A change is still backwards incompatible even if we've assured it won't break any currently observed users, and I don't think our docs should describe eg "deleting a field nobody uses" as backwards compatible.  We need a different term like safe or something.